extmod/btstack/btstack.mk

```diff
@@ -24,25 +28,30 @@ INC += -I$(BTSTACK_DIR)/3rd-party/yxml
 SRC_BTSTACK = \
 	$(addprefix lib/btstack/src/, $(SRC_FILES)) \
 	$(addprefix lib/btstack/src/ble/, $(filter-out %_tlv.c, $(SRC_BLE_FILES))) \
-	lib/btstack/platform/embedded/btstack_run_loop_embedded.c \
+	lib/btstack/platform/embedded/btstack_run_loop_embedded.c
+
+ifeq ($(MICROPY_BLUETOOTH_BTSTACK_USB),1)
+SRC_BTSTACK += \
+	lib/btstack/platform/libusb/hci_transport_h2_libusb.c
+
+CFLAGS  += $(shell pkg-config libusb-1.0 --cflags)
+LDFLAGS += $(shell pkg-config libusb-1.0 --libs)
+endif
 
-ifeq ($MICROPY_BLUETOOTH_BTSTACK_ENABLE_CLASSIC,1)
+ifeq ($(MICROPY_BLUETOOTH_BTSTACK_ENABLE_CLASSIC),1)
 include $(BTSTACK_DIR)/src/classic/Makefile.inc
 SRC_BTSTACK += \
 	$(addprefix lib/btstack/src/classic/, $(SRC_CLASSIC_FILES))
 endif
 
-SRC_LIB += $(SRC_BTSTACK)
-
-#$(BUILD)/lib/btstack/src/classic/btstack_link_key_db_static.o: CFLAGS += -Wno-error=pointer-arith
-
-# Incorrect %u, should be %lu.
-$(BUILD)/lib/btstack/src/classic/a2dp_source.o: CFLAGS += -Wno-error=format=
-$(BUILD)/lib/btstack/src/classic/btstack_sbc_decoder_bluedroid.o: CFLAGS += -Wno-error=format=
-$(BUILD)/lib/btstack/src/classic/btstack_link_key_db_tlv.o: CFLAGS += -Wno-error=format=
-$(BUILD)/lib/btstack/src/classic/goep_client.o: CFLAGS += -Wno-error=format=
-$(BUILD)/lib/btstack/src/ble/le_device_db_tlv.o: CFLAGS += -Wno-error=format=
-
+LIB_SRC_C += $(SRC_BTSTACK)
 
+# Suppress some warnings (e.g. %lu vs %u, unused variables, etc).
+BTSTACK_WARNING_CFLAGS = -Wno-error=old-style-definition -Wno-error=unused-variable -Wno-error=unused-parameter
+ifneq ($(CC),clang)
+BTSTACK_WARNING_CFLAGS += -Wno-error=format=
```

Comment:
I didn't see a need for this, not in unix coverage, dev or 32-bit coverage.  Do you know where it's needed?

Reply:
It's needed for a few things in the BT Classic code, and also for le_device_db_tlv.c

Neither are currently included in the build, but I'll likely have to add this back later.

Reply:
Ok!